### ISSUE TITLE
Cannot Delete Images Using Image ID After VCH Restart

### ISSUE THREAD

**Issue by gigawhitlocks** [vmware/vic#6076]
As a user I should be able to do `docker images` and then use an Image ID from that output to delete images. Likewise, doing `docker rmi -f $(docker images -aq)` should successfully remove all images present on the VCH.

Current behavior:
1) Install VCH
2) Pull some images (examples below)
3) Restart VCH
4) Follow below:
```bash
~/go/src/github.com/vmware/vic fix-local-build-issues*
❯ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
ubuntu              latest              ccc7a11d65b1        11 days ago         120MB
tomcat              latest              72d2be374029        12 days ago         292MB
nginx               latest              b8efb18f159b        3 weeks ago         107MB
golang              latest              6ce094895555        3 weeks ago         699MB
python              latest              968120d8cbe8        4 weeks ago         684MB
ruby                latest              3630c02d3d1b        4 weeks ago         679MB
fedora              latest              49236bc2f0da        4 weeks ago         231MB
alpine              latest              7328f6f8b418        7 weeks ago         3.96MB

~/go/src/github.com/vmware/vic fix-local-build-issues*
❯ docker rmi ccc7a11d65b1
Error response from daemon: No such image: ccc7a11d65b1:latest
```
Desired behavior:
```
[vagrant@devbox:/opt/go/src/github.com/vmware/vic(reload)] docker rmi ccc7a11d65b1
Untagged: ubuntu:latest
Untagged: ubuntu@sha256:34471448724419596ca4e890496d375801de21b0e67b81a77fd6155ce001edad
Deleted: sha256:ccc7a11d65b1b5874b65adb4b2387034582d08d65ac1817ebc5fb9be1baa5f88
Deleted: sha256:cb5450c7bb149c39829e9ae4a83540c701196754746e547d9439d9cc59afe798
Deleted: sha256:364dc483ed8e64e16064dc1ecf3c4a8de82fe7f8ed757978f8b0f9df125d67b3
Deleted: sha256:4f10a8fd56139304ad81be75a6ac056b526236496f8c06b494566010942d8d32
Deleted: sha256:508ceb742ac26b43bdda819674a5f1d33f7b64c1708e123a33e066cb147e2841
Deleted: sha256:8aa4fcad5eeb286fe9696898d988dc85503c6392d1a2bd9023911fb0d6d27081
[vagrant@devbox:/opt/go/src/github.com/vmware/vic(reload)]
```

Workaround available: delete the image with the human-readable name, e.g. `ubuntu:latest`

Acceptance Criteria:
Fix above behavior *and add an integration test* as it seems we currently only test deleting `human-readable-name:tag`and this is likely a regression.

**Comment by gigawhitlocks** [comment]
Two new bits of information:
1) @caglar10ur was able to reproduce this error simply by restarting the docker personality, so the issue is likely related to the way that the docker personality rebuilds its knowledge of what images are present
2) I was able to also reproduce this issue in v1.1.1 of VIC, so it does not appear to be a regression:

```
~/go/src/github.com/vmware/vic master*
❯ docker pull nginx:latest
Pulling from library/nginx
94ed0c431eb5: Pull complete 
a3ed95caeb02: Pull complete 
9406c100a1c3: Pull complete 
aa74daafd50c: Pull complete 
Digest: sha256:788fa27763db6d69ad3444e8ba72f947df9e7e163bad7c1f5614f8fd27a311c3
Status: Downloaded newer image for library/nginx:latest

~/go/src/github.com/vmware/vic master*
❯   restart () {
      govc vm.power -reset=true $(govc ls /ha-datacenter/vm | grep ian-1.1.1-vch) \
          && \
          while ! (docker info 2>/dev/null); do
              echo -n .;
          done
  }

~/go/src/github.com/vmware/vic master*
❯ restart
Reset VirtualMachine:2766... OK
.............................Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: v1.1.1-10711-56a309f
Storage Driver: vSphere Integrated Containers v1.1.1-10711-56a309f Backend Engine
VolumeStores: 
vSphere Integrated Containers v1.1.1-10711-56a309f Backend Engine: RUNNING
 VCH CPU limit: 12559 MHz
 VCH memory limit: 66.95 GiB
 VCH CPU usage: 412 MHz
 VCH memory usage: 20.38 GiB
 VMware Product: VMware ESXi
 VMware OS: vmnix-x86
 VMware OS version: 6.5.0
Plugins: 
 Volume: 
 Network: bridge
Swarm: inactive
Operating System: vmnix-x86
OSType: vmnix-x86
Architecture: x86_64
CPUs: 12559
Total Memory: 66.95GiB
ID: vSphere Integrated Containers
Docker Root Dir: 
Debug Mode (client): false
Debug Mode (server): false
Registry: registry-1.docker.io
Experimental: false
Live Restore Enabled: false


~/go/src/github.com/vmware/vic master* 16s
❯ docker rmi -f $(docker images -aq)
Error response from daemon: No such image: b8efb18f159b:latest
```

**Comment by anchal-agrawal** [comment]
Added kind/note - @stuclem we should release-note this if it's not fixed in 1.2.

**Comment by stuclem** [comment]
An attempt at a release note:

------

- <b>Cannot delete images using image ID after restarting VCH. [#6076](https://github.com/vmware/vic/issues/6076)</b><br>If you pull images into a VCH, then restart that VCH, running `docker rmi image_ID` results in `Error response from daemon: No such image`.

   Workaround: Delete images by using the human-readable name.

------

@gigawhitlocks is this OK? Thanks!

**Comment by gigawhitlocks** [comment]
Yup that's the gist of it @stuclem

**Comment by stuclem** [comment]
Thanks @gigawhitlocks

**Comment by stuclem** [comment]
@anchal-agrawal did this fix go into 1.3.1? Thanks!

**Comment by anchal-agrawal** [comment]
Yes.
